transforms/csdl/preprocess_csdl.xsl

```diff
@@ -695,6 +695,31 @@
         </xsl:copy>
     </xsl:template>
 
+    <!-- Add paths for user mailboxSettings by adding annotations to read and update the complex property-->
+    <xsl:template match="edm:Schema[@Namespace='microsoft.graph']/edm:EntityType[@Name='user']/edm:Property[@Name='mailboxSettings']">
+        <xsl:copy>
+            <xsl:apply-templates select="@* | node()"/>
+            <xsl:element name="Annotation">
+                <xsl:attribute name="Term">Org.OData.Capabilities.V1.ReadRestrictions</xsl:attribute>
+                <xsl:element name="Record" namespace="{namespace-uri()}">
+                    <xsl:element name="PropertyValue">
+                        <xsl:attribute name="Property">Readable</xsl:attribute>
+                        <xsl:attribute name="Bool">true</xsl:attribute>
+                    </xsl:element>
+                </xsl:element>
+            </xsl:element>
+            <xsl:element name="Annotation">
```

Comment:
We can re-use the `UpdateRestrictionsTemplate`
<xsl:call-template name="UpdateRestrictionsTemplate">
    <xsl:with-param name="updatable">true</xsl:with-param>
</xsl:call-template>

Reply:
Thanks. For some reason I thought the templates were for PUTs